src/post/crash-post-client.ts

```diff
@@ -0,0 +1,100 @@
+import { BugSplatApiClient, BugSplatFile, BugSplatResponse, Environment, S3ApiClient } from '@common';
+import { CrashType } from '@post';
+
+export class CrashPostClient {
+
+    private _processorApiClient: BugSplatApiClient;
+    private _s3ApiClient: S3ApiClient;
+
+    constructor(
+        private _database: string,
+        private _environment: Environment = Environment.Node,
+        private _processorIpAddress: string = ''
+    ) {
+        this._processorApiClient = new BugSplatApiClient(
+            `https://${this._database}.bugsplat.com`,
+            this._environment
+        );
+        this._s3ApiClient = new S3ApiClient();
+    }
+
+    async postCrash(
+        application: string,
+        version: string,
+        type: CrashType,
+        file: BugSplatFile,
+        md5 = ''
+    ): Promise<BugSplatResponse> {
+        const uploadUrl = await this.getCrashUploadUrl(
+            this._database,
+            application,
+            version,
+            file.size
+        );
+
+        await this._s3ApiClient.uploadFileToPresignedUrl(uploadUrl, file);
+
+        return this.commitS3CrashUpload(
+            uploadUrl,
+            this._database,
+            application,
+            version,
+            type,
+            md5,
+            this._processorIpAddress
+        );
+    }
+
+    private async getCrashUploadUrl(
+        database: string,
+        application: string,
+        version: string,
+        size: number
+    ): Promise<string> {
+        const route = 'api/getCrashUploadUrl.php'
+            + `?database=${database}`
+            + `&appName=${application}`
+            + `&appVersion=${version}`
+            + `&crashPostSize=${size}`;
+        const response = await this._processorApiClient.fetch(route);
+        if (response.status !== 200) {
+            throw new Error('Failed to get crash upload url!');
+        }
+
+        const json = await response.json();
+        return json.url;
+    }
+
+    private async commitS3CrashUpload(
+        s3Key: string,
+        database: string,
+        application: string,
+        version: string,
+        crashType: CrashType,
+        md5: string,
+        ipAddress?: string,
+    ): Promise<BugSplatResponse> {
+        const route = '/api/commitS3CrashUpload.php';
+        const formData = this._processorApiClient.createFormData();
+        formData.append('database', database);
+        formData.append('appName', application);
+        formData.append('appVersion', version);
+        formData.append('crashType', crashType);
+        formData.append('s3key', s3Key);
+        formData.append('md5', md5);
+
+        if (ipAddress) {
+            const processor = `OBAN-${ipAddress}`;
```

Comment:
Let's not append this prefix in code, let's make it so that it's provided in the constructor